### PR TITLE
Add support to change numeric options in settings screen

### PR DESCRIPTION
**Pull request by cgzones**
Like `delay` or `highlightDelaySecs`

### CODE REVIEW

File: OptionItem.c [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+   this->max = max;
+   return this;
+}
+
+NumberItem* NumberItem_newByVal(const char* text, int value, int min, int max) {
+   assert(min <= max);
+
+   NumberItem* this = AllocThis(NumberItem);
+   this->text = xStrdup(text);
+   this->value = value;
```
Comment (BenBE) [inline OptionItem.c:138]: Force min/max clamping on ctor?

File: OptionItem.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#define OptionItem_kind(this_)              As_OptionItem(this_)->kind
+
+typedef struct OptionItem_ {
+   Object super;
+} OptionItem;
+
+typedef struct CheckItem_ {
+   OptionItem super;
+
+   char* text;
```
Comment (BenBE) [inline OptionItem.h:36]: Put `text` in `OptionItem` as common field?

File: OptionItem.c [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+   free(this);
+}
+
+static void NumberItem_display(const Object* cast, RichString* out) {
+   const NumberItem* this = (const NumberItem*)cast;
+   assert (this != NULL);
+
+   char buffer[12];
+   RichString_write(out, CRT_colors[CHECK_BOX], "[");
+   xSnprintf(buffer, sizeof(buffer), "%d", NumberItem_get(this));
```
Comment (BenBE) [inline OptionItem.c:53]: Allow for sclaed display?

E.g. for the refresh delay it stores 1 internally, but shows 0.1 using a provided scale factor?

**Review by BenBE — COMMENTED** [review]
Looks good so far.

File: OptionItem.c (hunk 1 above)
Comment (cgzones) [reply OptionItem.c]: added

File: OptionItem.h (hunk 2 above)
Comment (cgzones) [reply OptionItem.h]: changed

File: OptionItem.c (hunk 3 above)
Comment (cgzones) [reply OptionItem.c:45]: then it gets complicated, e.g. how to display `0.1` as it is no exact floating point number.

Comment (BenBE) [reply OptionItem.c:45]: Two options:
1. Just use a fixed point display (number of digits behind decimal point provided), display with zero-padding at least `fixed-point+1` digits, insert decimal point at correct place.
2. `xSnprintf("%.*f", decimal_places, (double)number/scale);` (cf. https://stackoverflow.com/a/16413665)
3. Allow for custom display handler to be provided (probably overkill)

Option 2 is probably fully sufficient and easiest to implement.

Comment (cgzones) [reply OptionItem.c:45]: Support for scaling in both directions (`*10`, `/10`) and scaling with non-multiples of 10 will also complicate things.
d
After all is this really needed or can we expect user who want to change the scan interval what deci-seconds are?

Comment (BenBE) [reply OptionItem.c:45]: I think it's more intuitive, if the user sees fractional seconds. I don't think deciseconds is common enough that the average user will understand without mentally stumpbling over it first.

Scaling with just `*10` and `/10` (or powers thereof) should suffice (and works as suggested above).

**Comment by cgzones** [comment]
Note: when merging, check for new introduced configuration settings and remove redundant `xstrdup` calls in first arguments of `CheckItem_newByRef/CheckItem_newByValue`. (e.g. #305)

**Review by BenBE — APPROVED** [review]
Space with `-H10000` one too little, but still acceptable.

**Pull request merged**